ch_pipeline/analysis/calibration.py

```diff
@@ -2364,3 +2365,65 @@ def _calculate_uv(freq, prod, inputmap):
     uv = dist[:, np.newaxis, :] / lmbda[np.newaxis, :, np.newaxis]
 
     return uv
+
+class ExtractGaltAutoCorrelation(task.SingleTask):
+    """Extract the autocorrelations of the Galt telescope from a holography acquisition."""
+
+    def process(self, data):
+        """Extract the Galt autocorrelations and write them to disk.
+        Parameters
+        ----------
+        data: TimeStream
+            A TimeStream container holding a raw holography acquisition.
+        Returns
+        -------
+        autocorrelation: containers.GaltAutocorrelation
+            A GaltAutocorrelation container holding the extracted Galt autos
+            as a function of frequency, polarization product, and time.
+        """
+        # Redistribute over freq
+        data.redistribute("freq")
+
+        # Locate the holographic indices
+        layout_graph = layout.graph.from_db(data.time[0])
+
+        galt_inputs = get_holographic_index(
+            get_correlator_inputs(layout_graph, correlator="chime")
+        )
+
+        # Get the product map and inputs
+        prodmap = data.prod
+        ina, inb = prodmap["input_a"], prodmap["input_b"]
+
+        # Locate the Galt autocorrelations and cross-pol correlation
+        flag_YY = np.where((ina == galt_inputs[0]) & (inb == galt_inputs[0]), 1, 0)
+        flag_YX = np.where((ina == galt_inputs[0]) & (inb == galt_inputs[1]), 1, 0)
+        flag_XX = np.where((ina == galt_inputs[1]) & (inb == galt_inputs[1]), 1, 0)
+
+        auto_flag = (flag_YY + flag_YX + flag_XX).astype(bool)
```

Comment:
I think the way this is done, there is no guarantee that the products picked out by `auto_flag` will have the polarisation order that you define when creating the output container (although in practice it may end up working out based on the way the CHIME holography acquisitions are ordered).

I would probably construct the polarisation axis as follows:

```
pol = np.array([input_map[aa].pol + input_map[bb].pol for aa, bb in prodmap[auto_flag]])